support-workers/src/main/scala/com/gu/support/workers/lambdas/CreatePaymentMethod.scala

```diff
@@ -20,38 +20,51 @@ class CreatePaymentMethod(servicesProvider: ServiceProvider = ServiceProvider)
 
   def this() = this(ServiceProvider)
 
-  override protected def servicesHandler(state: CreatePaymentMethodState, requestInfo: RequestInfo, context: Context, services: Services) = {
+  override protected def servicesHandler(state: CreatePaymentMethodState, requestInfo: RequestInfo, context: Context, services: Services): FutureHandlerResult = {
     SafeLogger.debug(s"CreatePaymentMethod state: $state")
-    createPaymentMethod(state, services)
-      .map(paymentMethod =>
-        HandlerResult(
-          getCreateSalesforceContactState(state, paymentMethod),
-          requestInfo
-            .appendMessage(s"Payment method is ${paymentMethod.toFriendlyString}")
-            .appendMessage(s"Product is ${state.product.describe}")
-        ))
+    state.paymentFields.map(
+      paymentFields =>
+        createPaymentMethod(paymentFields, state.user, state.product.currency, services)
+        .map(paymentMethod =>
+          HandlerResult(
+            getCreateSalesforceContactState(state, Some(paymentMethod)),
+            requestInfo
+              .appendMessage(s"Payment method is ${paymentMethod.toFriendlyString}")
+              .appendMessage(s"Product is ${state.product.describe}")
+          ))
+    ).getOrElse(
+      Future.successful(HandlerResult(
+        getCreateSalesforceContactState(state, None),
+        requestInfo
+          .appendMessage(s"No Payment method present")
+          .appendMessage(s"Product is a corporate Digital Subscription")
```

Comment:
> Product is a corporate Digital Subscription

is it? looks like all we checked that it's not got payment fields (back to my comments about modelling with ADTs)

Reply:
if we start accidentally putting through other things without payment fields, due to a bug, this could confuse people looking into it.